discovery/memberlist.go

```diff
@@ -245,7 +245,7 @@ func NewLogWriter(log logrus.FieldLogger) *io.PipeWriter {
 	return writer
 }
 
-func split(addr string) (string, int, error) {
+func split(addr string) (retHost string, retPort int, reterr error) {
 	host, port, err := net.SplitHostPort(addr)
```

Comment:
I needed to differentiate working values from return values.  I avoid reassigning named return values.  This would avoid potential issues where return values contain unexpected state on certain edge conditions because a value was set temporarily then returned.

Reply:
And the reason for the named return values was, of course, lint.  I thought the original code was decent, so would prefer not to refactor it.

Reply:
OK